scipy/spatial/kdtree.py

```diff
@@ -222,6 +222,9 @@ class KDTree(cKDTree):
         is the boxsize along i-th dimension. The input data shall be wrapped
         into :math:`[0, L_i)`. A ValueError is raised if any of the data is
         outside of this bound.
+    build_workers : int, optional
```

Comment:
We better check with @peterbell10, @rainwoodman, and @sturlamolden about the API here.

Plus, they are each capable of reviewing C++ changes.

Another question might be related to `asv` benchmarks/testing with > 1 thread, but probably best to see what they say initially.

Reply:
Just to be clear. On my side I am just commenting on the interface to make it coherent with such existing options.

Reply:
> Another question might be related to asv benchmarks/unit testing with > 1 thread, but probably best to see what they say initially.

I agree with this,. but I do not know how to add such tests. Do you have any pointers?

Reply:
I added a test dimensionwith `[1, 2]` workers [here](https://github.com/scipy/scipy/pull/14481/files#diff-5c9e7e8948e7ff6efc82ab31a7edc2f9cc09abcd130808520be342f4afa76668)